src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java

```diff
@@ -1,11 +1,162 @@
+    private final WPI_TalonFX turretMotor;
+    private final WPI_TalonFX hoodMotor;
+
+    // Constants
+    public static final double FLYWHEEL_VELOCITY = 10;
+    public static final double STOP_MOTOR = 0;
+    public static final double DEGREES_TO_ENCODER_TICKS = 2048 / 360; // 2048 ticks per 360 degrees
+    public static final double MIN_TURRET_ANGLE = -180; // Total ~360 degrees of rotation, assumes 0 is center
+    public static final double MAX_TURRET_ANGLE = 180;
+    public static final double TURRET_MAX_SPEED = 0.1;
+    public static final double TURRET_MIN_SPEED = -0.1;
+    public static final double TURRET_P = 0.01; // Placeholder PID constants
+    public static final double TURRET_I = 0;
+    public static final double TURRET_D = 0;
+    public static final double MAX_HOOD_ANGLE = 40.0;
+    public static final double MIN_HOOD_ANGLE = 5;
+    public static final SupplyCurrentLimitConfiguration flywheelCurrentLimit = new SupplyCurrentLimitConfiguration(true,
+            40, 40, 500);
+    public static final SupplyCurrentLimitConfiguration turretCurrentLimit = new SupplyCurrentLimitConfiguration(true,
+            10, 10, 500);
+    public static final SupplyCurrentLimitConfiguration hoodCurrentLimit = turretCurrentLimit;
+
+    /**
+     * Constructor for shooter subsystem.
+     * 
+     * @param flywheelMotor1
+     *            The first motor connected to the flywheel
+     * 
+     * @param flywheelMotor2
+     *            The second motor connected to the flywheel
+     * 
+     * @param turretMotor
+     *            The motor that controls the horizontal rotation of the
+     *            turret
+     * 
+     * @param hoodMotor
+     *            The motor that controls the angle of the hood above the
+     *            turret
+     * 
+     */
+    public ShooterSubsystem(WPI_TalonFX flywheelMotor1, WPI_TalonFX flywheelMotor2, WPI_TalonFX turretMotor,
+            WPI_TalonFX hoodMotor) {
+        // Motor configs
+        flywheelMotor1.configSupplyCurrentLimit(flywheelCurrentLimit);
+        flywheelMotor1.setNeutralMode(NeutralMode.Coast);
+        flywheelMotor2.configSupplyCurrentLimit(flywheelCurrentLimit);
+        flywheelMotor2.setNeutralMode(NeutralMode.Coast);
+        flywheelMotor1.setInverted(false);
+        flywheelMotor2.follow(flywheelMotor1);
+        flywheelMotor2.setInverted(InvertType.OpposeMaster);
+
+        turretMotor.configForwardSoftLimitThreshold(MAX_TURRET_ANGLE * DEGREES_TO_ENCODER_TICKS);
+        turretMotor.configReverseSoftLimitThreshold(MIN_TURRET_ANGLE * DEGREES_TO_ENCODER_TICKS);
+        turretMotor.configForwardSoftLimitEnable(true);
+        turretMotor.configReverseSoftLimitEnable(true);
+        turretMotor.configSupplyCurrentLimit(turretCurrentLimit);
+        turretMotor.setNeutralMode(NeutralMode.Brake);
+        turretMotor.configSelectedFeedbackSensor(TalonFXFeedbackDevice.IntegratedSensor, 0, 0);
+        turretMotor.config_kP(0, TURRET_P);
```

Comment:
kP needs to be non-zero for the motor to do anything at all.  I'd recommend starting small (say, 0.01).

Reply:
We'll also need to tune the flywheelMotor PID values, but leaving them at the defaults is fine for now.

Reply:
I'm confused about your first comment... `config_kP` has parameters `slotIdx` and `value` in that order ([BaseMotorController.config_kP()](https://store.ctr-electronics.com/content/api/java/html/classcom_1_1ctre_1_1phoenix_1_1motorcontrol_1_1can_1_1_base_motor_controller.html#ae8fec79c2b494cef36876936ddf44425)), so kP is getting set to `TURRET_P`, which is 0.01.

Reply:
Missed that the 2nd argument is the actual value, as there's no variable hinting in GitHub PR's viewer.